### ISSUE TITLE
Ideas in regards to preferences and usability

### ISSUE THREAD

**Issue by frgomes** [rastersoft/devedeng#12]
Recently I've faced difficulties with devedeng which actually were related to other software:
1. avconv was not accepting certain arguments anymore;
2. a bug in libav (and also in ffmpeg) prevents mp2 audio codec from working properly;

The first issue implied that devedeng needed a fix to accomodate the new command line interface.
The second issue implies that the user must _remember_ to explicitly select ac3 for the menu audio, otherwise avconv will fail when it tries to create the menu.

I suggest that the preferences dialog allows settings for profiles (such as DVD, VideoCD, etc).
This way, I could select ac3 for the menu and forget the bug in libav. It would also be useful to guarantee that I'm always creating a DVD in accordance with some settings I find optimal for my purposes.

Another suggestion is in regards to "Adjust disc usage".
My use case is: I know that 2 videos account for 120% and I'm happy to press "Adjust Disk Usage" even if that implies on some loss of quality. But I suppose that there are few more use cases here and I let you decide. The point is: I would be great if I would not have to press "Adjust Disk Usage" every time.

Thanks

**Comment by glenntanner3** [comment]
I'm working on an update to the settings which should allow you to preset all options.

As for the disc usage, that's a bit tougher as videos are subjective in duration and you may want to make one unique. However I could see making a prompt that when the videos are over 100% adjusting at that time. Would that be an acceptable adjustment?

I'm sorry you faced the problem with libav, I'm not sure how popular it is vs ffmpeg which it forked from. With all the DVDs that I create i'm not sure why I did not see the same with the 448 audio bit rate for ac3 bug, from what I looked up ac3 does not support 448 so there could be a bad UI setting that allows this to be set that needs to be looked into. 

Thank you for your help resolving the libav error.

**Comment by frgomes** [comment]
AFAIK, libav is used by Debian and derivatives (not sure exactly which derivatives).
Debian Jessie does not even offer ffmpeg via apt-get but this situation will change in the next release.

I found bug reports involving libav in Ubuntu, for example, so I imagine that libav is surfing on the popularity of Ubuntu. I only have Debian in my computers and I'm not sure if libav is the only option available via apt-get in Ubuntu. I will discover this Monday (cos they use Kubuntu in the office) and let you know.

In regards to "Adjust disk usage"... yes... I realize that it requires special care only when disk usage overuns 100%. _Maybe_ an additional step could be added if the user presses "Forward" without first deciding the matter in regards to disk space usage. Just a suggestion. But my oppinion does not represent much because this is the first time I use devede/devedeng and still only restricted to a specific use case.

Thanks

**Comment by frgomes** [comment]
In regards to ac3: I have to "remember" to select it instead of mp2 for the menu.
It worked properly for me, apparently... but I haven't checked the bitrate involved. I will take a look later on tonight when I will try some other things in regards to that clip #3/3 which is still causing troubles.

By the way, I've noticed that you fixed the maximum bitrate in case of mp2, only. Not ac3. But I haven't examined the code deeply. Better if I run again and inform what the bitrate is. I will do.

**Comment by glenntanner3** [comment]
Yeah, I didn't go through all the supported codecs but will... not to mention I put the check on the backend which the user will not know about, which is not fair to the user.

**Comment by frgomes** [comment]
This article below confirms what I believed in regards to Debian and Ubuntu pumping up popularity of libav. But, as I said, Debian is going to support ffmpeg in the next release. I simply do not have side on this war. I'm simply not willing to compile ffmpeg from sources for Debian.

http://blog.pkh.me/p/13-the-ffmpeg-libav-situation.html

**Comment by rastersoft** [comment]
Remember that DevedeNG has separated support for AVConv and FFMpeg...

**Comment by frgomes** [comment]
Yes. True. The problem is not devedeng, but bugs in libav. From my brief research I would say that ffmpeg would be beneficial to me but compiling from sources such a big and complicated thing is a price far higher than I would like to pay.

**Comment by frgomes** [comment]
I've ended up installing ffmpeg onto Debian Jessie and I was finally able to burn all DVDs. Even the troublesome clip #3 worked relatively easy.

I just had to cut the clip again and advance 1sec because apparently there was a "blank" of 4 or 5 secs in the video stream for some reason. Advancing in 1 sec the cut point was enough for ffmpeg determine the input format. I hope I didn't say too much stupidity in the last couple of sentences.

Given that everything worked with _ffmpeg_ and I used the default mp2 enconding for the menu, I was not able to collect the command involving ac3.
